src/screencast/screencast_common.c

```diff
@@ -41,11 +46,18 @@ static char *gbm_find_render_node() {
 	return render_node;
 }
 
-struct gbm_device *xdpw_gbm_device_create(void) {
+struct gbm_device *xdpw_gbm_device_create(struct xdpw_screencast_context *ctx) {
 	struct gbm_device *gbm;
+	drmDevicePtr main_dev;
 	char *render_node = NULL;
 
-	render_node = gbm_find_render_node();
+	drmGetDeviceFromDevId(ctx->main_device, /* flags */ 0, &main_dev);
```

Comment:
If this fails (because the device disappeared, or the `dev_t` value got mangled, etc), `main_dev` is undefined memory and will cause a crash in `gbm_find_render_node`.

Reply:
I didn't find drmGetDiveFromDevId documented in the local installed header file, but the one in [your MR to mesa](https://gitlab.freedesktop.org/mesa/drm/-/merge_requests/129) suggest, that we can check for a non negative return value for success.

Reply:
I'd rather have the additional check to log if we are using the main_device or not.

Reply:
I reworked this to try the main_device, then try a fallback on each tranche_target_device and if nothing succeeds here, we use the first render device (which we also need for the pre feedback case).